packages/http-client/src/ceramic-http-client.ts

```diff
@@ -136,7 +137,7 @@ export default class CeramicClient implements CeramicApi {
     if (!this._docmap[docIdStr]) {
       this._docmap[docIdStr] = await Document.load(docId, this._apiUrl, this.context, this._config)
     }
-    this._docmap[docIdStr].doctypeHandler = this.findDoctypeHandler(this._docmap[docIdStr].state.doctype)
+    this._docmap[docIdStr].doctypeConstructor = this._doctypes[this._docmap[docIdStr].state.doctype]
```

Comment:
This dance between doctype and document makes me sad.  It seems like it should be possible to have this set when the document is first constructed rather than up here in the client.  Anyway that's a bigger change and doesn't need to happen now, this is good cleanup either way

Reply:
Me too :/